File: src/org/usfirst/frc5293/prefs/util/DoublePref.java
package org.usfirst.frc5293.prefs.util;

import edu.wpi.first.wpilibj.Preferences;
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;

public class DoublePref extends Pref<Double> {
    public DoublePref(String key, Double defaultValue) {
        super(key, defaultValue);
    }

    @Override
    public Double get() {
        return Preferences.getInstance().getDouble(key, defaultValue);
    }

    @Override
    public void set(Double value) {
        Preferences.getInstance().putDouble(key, value);
    }
}

File: src/main/kotlin/org/usfirst/frc5293/framework/prefs/DoublePref.kt
package org.usfirst.frc5293.framework.prefs

import edu.wpi.first.wpilibj.Preferences
import org.usfirst.frc5293.framework.util.Logging

class DoublePref(key: String, default: Double) : Pref<Double>(key, default), Logging {

    override fun get(): Double {
        return Preferences.getInstance().getDouble(key, default)
    }

    override fun set(value: Double) {
        super.set(value)
        Preferences.getInstance().putDouble(key, value)
    }
}
